Yafc.UI/Core/DrawingSurface.cs

```diff
@@ -11,9 +11,9 @@ namespace Yafc.UI {
 
         public TextureHandle Destroy() {
             if (valid) {
-                SDL.SDL_DestroyTexture(handle);
+                var capturedHandle = handle;
```

Comment:
Is this var truly needed?

Reply:
Definitely. My understanding is that we really don't want to rely on `this` once we're inside the main thread callback, because once `Destroy()` is called the CLR is free to do whatever it wants with the freed-up memory. Creating this as a local lets us close over just the local, preventing that issue.

Reply:
Yep, it's definitely needed. Trying to remove it gives this error:

> error CS1673: Anonymous methods, lambda expressions, query expressions, and local functions inside structs cannot access instance members of 'this'. Consider copying 'this' to a local variable outside the anonymous method, lambda expression, query expression, or local function and using the local instead.

Reply:
Destroy isn't anything special that would allow `this` to be freed. You are capturing the handle inside the lambda and i think that would just resurrect the object until the lambda is gc'd. Mm indeed you might want to avoid that resurrection so i do agree with this var now i've read up on it more.

But the special thing you're probably thinking about is not Destroy, but the finalizer, recognized by the ~ in front of the method, like on line 95: https://github.com/shpaass/yafc-ce/blob/3cb07938c570e63d8e11ae217fcfe33a26d90e41/Yafc.UI/Core/DrawingSurface.cs#L95

I noticed this finalizer calls dispose, and dispose then calls SDL_DestroyRenderer. Possibly you want to also (or especially) wrap that one in a Dispatch?
https://github.com/shpaass/yafc-ce/blob/3cb07938c570e63d8e11ae217fcfe33a26d90e41/Yafc.UI/Core/DrawingSurface.cs#L87